python/src/equistore/operations/slice.py

```diff
@@ -327,21 +327,30 @@ def _slice_tensorblock(
     # create filters for Gradient TensorBlocks as there isn't a one-to-one mapping
     # between samples of the parent TensorBlock and its Gradients. This doesn't
     # need to be done for properties as the mapping is by definition a one-to-one.
-    numeric_idxs = np.array([i for i, val in enumerate(samples_filter) if val])
+    numeric_idxs = set(np.array([i for i, val in enumerate(samples_filter) if val]))
```

Comment:
I don't think you need to construct the np.array here
```suggestion
    numeric_idxs = set([i for i, val in enumerate(samples_filter) if val])
```

Reply:
Even better, you can use a generator to construct the set and also bypass the list:
```suggestion
    numeric_idxs = set(i for i, val in enumerate(samples_filter) if val)
```

Reply:
Change made in 294420dcdbfab3649434b39052e91ca10aaa9c61.